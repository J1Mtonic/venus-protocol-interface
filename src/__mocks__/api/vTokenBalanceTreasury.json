[
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "9999999989200000000000000000",
    "vToken": "0x2fF3d0F6990a40261c66E1ff2017aCBc282EB6d0"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "100570",
    "vToken": "0xecA88125a5ADbe82614ffC12D0DB554E2e2867C8"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "0",
    "vToken": "0xfD5840Cd36d94D7229439859C0112a4185BC0255"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "209682159264619399",
    "vToken": "0x95c78222B3D6e262426483D42CfA53685A67Ab9D"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "10570584835727719077872",
    "tokenBalance": "10570584835727719077872",
    "vToken": "0xA07c5b74C9B40447a954e1466938b865b6BBea36"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "0",
    "vToken": "0x151B1e2635A717bcDc836ECd6FbB62B674FE3E1D"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "0",
    "vToken": "0x882C173bC7Ff3b7786CA16dfeD3DFFfb9Ee7847B"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "0",
    "vToken": "0xf508fCD89b8bd15579dc79A6827cB4686A3592c8"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "0",
    "vToken": "0x57A5297F2cB2c0AaC9D554660acd6D385Ab50c6B"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "0",
    "vToken": "0xB248a295732e0225acd3337607cc01068e3b9c10"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "0",
    "vToken": "0x9A0AF7FDb2065Ce470D72664DE73cAE409dA28Ec"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "0",
    "vToken": "0xec3422Ef92B2fb59e84c8B02Ba73F1fE84Ed8D71"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "0",
    "vToken": "0x5c9476fcd6a4f9a3654139721c949c2233bbbbc8"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "0",
    "vToken": "0x86aC3974e2BD0d60825230fa6F355fF11409df5c"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "0",
    "vToken": "0x26DA28954763B92139ED49283625ceCAf52C6f94"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "0",
    "vToken": "0x08ceb3f4a7ed3500ca0982bcd0fc7816688084c3"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "0",
    "vToken": "0x61eDcFe8Dd6bA3c891CB9bEc2dc7657B3B422E93"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "0",
    "vToken": "0x78366446547D062f45b4C0f320cDaa6d710D87bb"
  },
  {
    "balanceOf": "0",
    "balanceOfUnderlying": "0",
    "borrowBalanceCurrent": "0",
    "tokenAllowance": "0",
    "tokenBalance": "0",
    "vToken": "0xb91A659E88B51474767CD97EF3196A3e7cEDD2c8"
  }
]
