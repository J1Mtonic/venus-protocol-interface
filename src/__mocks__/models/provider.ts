import type { Provider } from '@wagmi/core';
import { BigNumber } from 'ethers';

export const blockNumber = 123;
export const balance = BigNumber.from('1000000000000000000');

const getBlockNumber = vi.fn(async () => blockNumber);
const getBalance = vi.fn(async () => balance);

const provider = {
  getBlockNumber,
  getBalance,
} as unknown as Provider;

export default provider;
