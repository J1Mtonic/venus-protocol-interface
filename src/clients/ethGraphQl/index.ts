export { apolloClient } from './apolloClient';
export * from './gql';
export * from './gql/graphql';
