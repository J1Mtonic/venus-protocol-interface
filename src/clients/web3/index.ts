export { default as useMulticall } from './useMulticall';
export { default as useMulticall2 } from './useMulticall2';
export { default as Web3Wrapper } from './Web3Wrapper';
export * from './Web3Wrapper';
export * from './connectors';
export * from './types';
