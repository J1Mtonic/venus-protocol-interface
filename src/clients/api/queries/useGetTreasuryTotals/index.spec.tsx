import React from 'react';

import vTokenBalanceTreasury from '__mocks__/api/vTokenBalanceTreasury.json';
import { markets } from '__mocks__/models/markets';
import { getMarkets, useGetVTokenBalancesAll } from 'clients/api';
import renderComponent from 'testUtils/renderComponent';

import useGetTreasuryTotals, { UseGetTreasuryTotalsOutput } from '.';

jest.mock('clients/api');

describe('api/queries/useGetTreasuryTotals', () => {
  beforeEach(() => {
    (getMarkets as jest.Mock).mockImplementation(() => ({ markets }));

    (useGetVTokenBalancesAll as jest.Mock).mockImplementation(() => ({
      data: vTokenBalanceTreasury,
    }));
  });

  it('calculates totals correctly', async () => {
    let data: UseGetTreasuryTotalsOutput['data'];

    const CallMarketContext = () => {
      ({ data } = useGetTreasuryTotals());
      expect(data).toMatchSnapshot();
      return <div />;
    };

    renderComponent(<CallMarketContext />);
  });
});
