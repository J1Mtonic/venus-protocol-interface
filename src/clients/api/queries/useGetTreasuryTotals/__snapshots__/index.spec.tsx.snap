// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`api/queries/useGetTreasuryTotals > calculates totals correctly 1`] = `
{
  "treasuryBalanceCents": "62.049539256425566",
  "treasuryBorrowBalanceCents": "1717617594460750",
  "treasuryLiquidityBalanceCents": "37856027310",
  "treasurySupplyBalanceCents": "4198560158112936",
}
`;
